api/draft_registrations/permissions.py

```diff
@@ -23,8 +23,38 @@ def has_object_permission(self, request, view, obj):
             obj = obj.get('self', None)
         assert_resource_type(obj, self.acceptable_models)
         auth = get_user_auth(request)
+        if not auth:
+            return False
+
+        if isinstance(obj, DraftRegistration) and isinstance(obj.branched_from, Node):
+            obj = obj.branched_from
+
+        if request.method in permissions.SAFE_METHODS:
+            return obj.is_contributor(auth.user)
+        else:
+            return obj.can_edit(auth)
+
+class IsAdminContributor(permissions.BasePermission):
```

Comment:
Both permissions classes look good after your changes but can you update the docstrings?